user-guide/Advanced_Functionality/Databases/Indexing_database/Elasticsearch/storage_per_DMA/Profile_migration_to_elastic.md

```diff
@@ -0,0 +1,96 @@
+
+   - A window will show the migration actions that have been scheduled. If you close this window, the migration will continue in the background. To cancel the migration, click the *Cancel Migration* button.
+
+   - In the *MigrationStatus* table in SLNetClientTest tool, the *profile instances*, *definitions*, and *parameters* rows will reflect the progress of the migration.
+
+   - All profiles will be loaded from the *Profiles.xml* file on the DataMiner Agent where the migration was triggered. The *Profiles.xml* files of the other DataMiner Agents in the cluster are not migrated. This is not necessary, as the *Profiles.xml* file is synced in the cluster.
+
+   - The profiles objects will be written to the indexing database.
+
+   > [!IMPORTANT]
+   > A snapshot of the *Profiles.xml* file will be migrated, which means that Profile Manager is not stopped during the migration and keeps keep working as expected. However, **changes to profiles that are done during the migration will not be included** in the migration, so avoid creating, updating, or deleting profile objects until the migration is complete and the system has switched to using the indexing database for profile information.
+
+   > [!NOTE]
+   > The migration should not take more than a few minutes. During testing, migrating a *Profiles.xml* file of 100 MB in a system with a local Elasticsearch database took about 4 minutes.
+
+1. When the migration is complete, click the *Start using Elastic* button to switch to indexing database storage.
+
+   If the *Start using Elastic* button is not visible, that means there is no configuration file yet. in that case, click the *Create Configuration* button to create the configuration file.
+
+### Troubleshooting
+
+If the migration should fail for any reason, the migration status object in the SLNetClientTest tool window will get a red background color. The ``SLMigrationManager.txt`` and ``SLProfileManager.txt`` log files will contain more information.
+
+If a ``MigrationStatus`` is stuck in the ``InProgress`` state, you will need to cancel the migration and trigger the migration again. You can do so with the *Cancel Migration* button in the *ProfileManger XML to Elastic* section of the SLNetClientTest tool window.
+
+### Behavior in new installations
+
+When a new DataMiner Agent is installed, the used storage type will depend on when Profile Manager starts up for the first time.
+
+- If DataMiner is installed with the 10.0 installer and Profile Manager is used, XML storage will be used. An indexing database is not yet supported as a storage type for profiles in DataMiner 10.0. After you have upgraded this DataMiner Agent to DataMiner 10.0.8 or later, it will continue to use XML storage until you trigger the migration.
+
+- If DataMiner is installed on version 10.4.0 and Profile Manager is used, indexing database storage will be used. This means profiles will only be functional on a DataMiner System that has indexing database storage running.
```

Comment:
Do you mean if DataMiner is installed with the 10.4 installer (which is currently not yet available)? Or do you mean if it is installed with the 10.2 installer and upgraded to 10.4 before Profile Manager is first started? Or both?

Reply:
Yes both